python/lsst/daf/butler/formatters/fitsExposureFormatter.py

```diff
@@ -19,52 +19,106 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import os.path
-
-from lsst.daf.butler.formatters.fileFormatter import FileFormatter
+from ..core.formatter import Formatter
 
 __all__ = ("FitsExposureFormatter", )
 
 
-class FitsExposureFormatter(FileFormatter):
+class FitsExposureFormatter(Formatter):
     """Interface for reading and writing Exposures to and from FITS files.
     """
     extension = ".fits"
 
-    def _readFile(self, path, pytype):
-        """Read a file from the path in FITS format.
+    parameters = frozenset(("bbox", "origin"))
+
+    def read(self, fileDescriptor, comp=None):
+        """Read data from a file.
 
         Parameters
         ----------
-        path : `str`
-            Path to use to open the file.
-        pytype : `class`
-            Class to use to read the FITS file.
+        fileDescriptor : `FileDescriptor`
+            Identifies the file to read, type to read it into and parameters
+            to be used for reading.
+        comp : `str`, optional
```

Comment:
This looks like it's my fault because I call it `comp` in `fileFormatter`. posixDatastore does use that form of `formatter.read()`. You are right I forgot to change the abstract base class. The change here is to allow Exposure components to be read without requiring the entire Exposure to be read.

Reply:
I've renamed this to `component` and added it to the `Formatter` base class.